client/src/App.js

```diff
@@ -30,10 +31,23 @@ function App() {
   const [hidden, setHidden] = React.useState(false);
   const [innerHidden, setInnerHidden] = React.useState(false);
   const [isFullScreen, setIsFullScreen] = React.useState(false);
+  const [userData, setUserData] = React.useState(null);
   const toLightBox = () => {
     document.querySelector(".lightbox").scrollIntoView({ behavior: "smooth" });
   };
 
+  React.useEffect(() => {
+    axios({
+      method: "GET",
+      withCredentials: true,
+      url: "http://localhost:3001/user",
+    }).then((res) => {
+      setUserData(res.data);
+      //console.log(res.data); //brings side effects of affecting browser's performance -- infinity error
+      //console.log(userData.username) //brings side effects of affecting browser's performance -- infinity error
+    });
+  });
```

Comment:
What's happening here is that our axios call updates the `userData` state in its callback function, and that state update triggers a re-rendering of the component, which then triggers `userEffect` to run again, thus causing our infinite loop.

By adding an _empty_ dependency array as the second argument of `useEffect`, we are telling it to **only** run once after the initial rendering.

Here is a great description on this option -https://dmitripavlutin.com/react-useeffect-explanation/#2-the-dependencies-of-useeffect

Also he did a great writeup on the infinite loop error you're seeing as well - https://dmitripavlutin.com/react-useeffect-infinite-loop/

```suggestion
  }, []);
```

Reply:
Good thing to know. Thank you.